libraries/core_libs/network/src/tarcap/packets_handlers/dag_block_packet_handler.cpp

```diff
@@ -47,6 +47,8 @@ void DagBlockPacketHandler::process(const PacketData &packet_data, const std::sh
     }
 
     if (auto status = syncing_handler_->checkDagBlockValidation(block); !status.first) {
+      trx_mgr_->insertBroadcastedTransactions(new_transactions);
```

Comment:
maybe this should after `if (syncing_state_->is_pbft_syncing())` as we are still syncing we do not need those trxs

Reply:
I was also having doubts about that. If nodes are behaving honestly they should not be gossiping new blocks. This actually happens only when the node which sent this block thinks we are synced and this only happens when we are behind only by one block or so. So I think it is better to save transaction in this case